client/src/quit/components/QuitButton.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from "react";
+import {  Button, Header, Icon, Modal  } from "semantic-ui-react";
+import { handleSubmit } from "../api"
+
+
+class QuitButton extends React.Component{
+    public state = { modelOpen: false }
+
+    public handleOpen = () => {
+        this.setState({ modelOpen: true })
+    }
+
+    public handleClose = () => {
+        this.setState({ modelOpen: false })
+    }
+
+    public handleQuit = () => {
+        this.handleClose()
+        handleSubmit()
+    }
+
+    public render(){
+    return(
+        <Modal
+            trigger={<Button content="Quit" icon="shutdown" onClick={this.handleOpen} labelPosition="left" floated="right" />} 
+            open={this.state.modelOpen}
+            onClose={this.handleClose}
+            size='mini'
+        >
+            <Header icon='shutdown' content='Confirm Exit' />
+            <Modal.Content>
+                <p>
+                    Do you want to exit ?
+                </p>
+            </Modal.Content>
+            <Modal.Actions>
+                <Button color='red' onClick={this.handleClose}>
+                    <Icon name='remove'/> No
+                </Button>
+                <Button color='green' onClick={this.handleQuit} >
```

Comment:
If the user confirms exit by pressing "Yes" button, the request is sent to the server, and the modal closes. But there is delay in stopping the server.

Reply:
A suggestion: you might like implementing your Modal with React's functional components along with hooks.

Reply:
> there is delay in stopping the server

That's to be expected, as it takes a while for the worker processes to close down.

Reply:
Is there a chance of confusion for the user as he/she press "Yes" and the server is still running?

Reply:
> Is there a chance of confusion for the user as he/she press "Yes" and the server is still running?

What you can also do: after confirming the shutdown, turn the modal dialog into a spinner, like you get when the GUI runs but the server is shutdown. That way, the user will know that something is happening, and they will also know when the executor has finished shutting down.

Reply:
Instead of modal dialog, "shutdown" button will turn into loading indicator.

Reply:
Okay, that works, too! I think the button needs to be disabled when a shutdown is running, too, as otherwise, `handleQuit` is called multiple times